File: app/src/main/java/knf/kuma/favorite/FavoriteAdapter.java
package knf.kuma.favorite;

import android.content.Context;
import android.preference.PreferenceManager;
import android.support.annotation.LayoutRes;
import android.support.v4.app.Fragment;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.animeinfo.ActivityAnime;
import knf.kuma.commons.PicassoSingle;
import knf.kuma.pojos.FavoriteObject;

public class FavoriteAdapter extends RecyclerView.Adapter<FavoriteAdapter.ItemHolder> {

    private Context context;
    private Fragment fragment;
    private RecyclerView recyclerView;
    private List<FavoriteObject> list = new ArrayList<>();

    public FavoriteAdapter(Fragment fragment, RecyclerView recyclerView) {
        this.fragment = fragment;
        this.context = fragment.getContext();
        this.recyclerView = recyclerView;
    }

    @Override
    public ItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ItemHolder(LayoutInflater.from(parent.getContext()).inflate(getLayout(), parent, false));
    }

    @LayoutRes
    private int getLayout(){
        if (PreferenceManager.getDefaultSharedPreferences(context).getString("lay_type","0").equals("0")){
            return R.layout.item_fav;
        }else {
            return R.layout.item_fav_grid;
        }
    }

    @Override
    public void onBindViewHolder(final ItemHolder holder, int position) {
        final FavoriteObject object=list.get(position);
        PicassoSingle.get(context).load(object.img).into(holder.imageView);
        holder.title.setText(object.name);
        holder.type.setText(object.type);
        holder.cardView.setOnClickListener(view -> ActivityAnime.open(fragment, object, holder.imageView));
    }

    public void updateList(List<FavoriteObject> list){
        this.list=list;
        recyclerView.post(() -> notifyDataSetChanged());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ItemHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.card)
        CardView cardView;
        @BindView(R.id.img)
        ImageView imageView;
        @BindView(R.id.title)
        TextView title;
        @BindView(R.id.type)
        TextView type;

        public ItemHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/knf/kuma/favorite/FavoriteAdapter.kt
package knf.kuma.favorite

import android.content.Context
import android.preference.PreferenceManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.LayoutRes
import androidx.cardview.widget.CardView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import knf.kuma.R
import knf.kuma.animeinfo.ActivityAnime
import knf.kuma.commons.PatternUtil
import knf.kuma.commons.PicassoSingle
import knf.kuma.pojos.FavoriteObject
import java.util.*

class FavoriteAdapter(private val fragment: Fragment, private val recyclerView: RecyclerView) : RecyclerView.Adapter<FavoriteAdapter.ItemHolder>() {

    private val context: Context? = fragment.context
    private var list: MutableList<FavoriteObject> = ArrayList()

    private val layout: Int
        @LayoutRes
        get() = if (PreferenceManager.getDefaultSharedPreferences(context).getString("lay_type", "0") == "0") {
            R.layout.item_fav
        } else {
            R.layout.item_fav_grid
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemHolder {
        return ItemHolder(LayoutInflater.from(parent.context).inflate(layout, parent, false))
    }

    override fun onBindViewHolder(holder: ItemHolder, position: Int) {
        val favoriteObject = list[position]
        PicassoSingle[context!!].load(PatternUtil.getCover(favoriteObject.aid!!)).into(holder.imageView)
        holder.title.text = favoriteObject.name
        holder.type.text = favoriteObject.type
        holder.cardView.setOnClickListener { ActivityAnime.open(fragment, favoriteObject, holder.imageView) }
    }

    fun updateList(list: MutableList<FavoriteObject>) {
        this.list = list
        recyclerView.post { notifyDataSetChanged() }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ItemHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @BindView(R.id.card)
        lateinit var cardView: CardView
        @BindView(R.id.img)
        lateinit var imageView: ImageView
        @BindView(R.id.title)
        lateinit var title: TextView
        @BindView(R.id.type)
        lateinit var type: TextView

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}
